manifests/prepare.pp

```diff
@@ -52,14 +54,103 @@
     }
   }
 
-  # TODO: manage server.cfg contents
-  if $::mcollective_configured {
-    file { $::agent_upgrade::params::mcodirs:
-      ensure => directory,
+  # manage client.cfg and server.cfg contents
+  file { $::agent_upgrade::params::mcodirs:
+    ensure => directory,
+  }
+  # The mco_*_config facts will return the location of mcollective config (or nil), prefering PE over FOSS.
+  if $::mco_server_config {
+    $mco_server = $::agent_upgrade::params::mco_server
+    file { $mco_server:
+      ensure  => file,
+      source  => $::mco_server_config,
+    }
+
+    if $::mco_server_settings {
+      $mco_server_libdir = $::mco_server_settings['libdir']
+      if $mco_server_libdir {
+        $mco_server_libdir_array = split($mco_server_libdir, $::agent_upgrade::params::path_separator)
+        # Only add the new path if it's not already in libdir
+        if [] == $mco_server_libdir_array.filter |$x| { $x == $::agent_upgrade::params::mco_libdir } {
```

Comment:
I thought about using puppet's [match](https://docs.puppetlabs.com/references/4.1.latest/function.html#match) function to check if the new path is already in libdir (instead of splitting and filtering). But it gets a little complicated. If newpath was `/a`, then you wouldn't want to match `/a/b`, and you'd have to handle cases where it's at the start, in the middle, at the end, or not present.